src/goto-symex/symex_target_equation.cpp

```diff
@@ -966,6 +966,8 @@ void symex_target_equationt::SSA_stept::validate(
     validate_full_expr(ssa_lhs, ns, vm);
     validate_full_expr(ssa_full_lhs, ns, vm);
     validate_full_expr(original_full_lhs, ns, vm);
+    if(type == goto_trace_stept::typet::DECL)
+      break;
```

Comment:
In this case I'd prefer some copy&paste and just completely separate the `ASSIGNMENT` and `DECL` cases.

Reply:
+1

Reply:
done